Automattic/jetpack
# RNA components: Add Connection Status Card

#### Changes proposed in this Pull Request:
* The RNA Connection Status Card component.
* ~~Move the `/connection/data` endpoint from Jetpack to Connection package.~~ Will be handled in a separate PR

## Component `ConnectionStatusCard`
The `ConnectionStatusCard` component displays the current site and user connection status as well as the corresponding actions.
This component is meant to be used when at least the site level connection has been established, aka there's a status to display.
In cases where the site level connection has not been established yet, please use the `ConnectScreen` component instead.

It consists of 2 main states: 
1. *User account connected*: In this state the user will be presented with information regarding their connected WordPress.com account, aka the `username` and `avatar`.
2. *User account not connected*: In this state the user will be presented with an alert message informing them their account is not connected and a corresponding button that upon clicking will trigger the User Connection flow.

In all cases, users are presented with a link to `Disconnect` (see `DisconnectDialog` component).


### Properties
- *apiRoot* - string (required), API root URL.
- *apiNonce* - string (required), API Nonce.
- *isRegistered* - boolean (required), Whether a site level connection has already been established. If not, the component will not render.
- *isUserConnected* - boolean (required), Whether the current user has connected their WordPress.com account.
- *redirectUri* - string (required), The redirect admin URI after the user has connected their WordPress.com account.
- *title* - string, The Card title. Defaults to "Connection".
- *connectionInfoText* - string, The text that will be displayed under the title, containing info how to leverage the connection. Defaults to "Leverages the Jetpack Cloud for more features on your side."
- *onDisconnected* - callback, The callback to be called upon disconnection success.

### Important Notes
The `Disconnect` functionality is **temporary**. In the future, it will be replaced with a link to a central connection management page from where users will be able to disconnect.


### Basic Usage
```jsx
import React, { useCallback } from 'react';
import { ConnectionStatusCard } from '@automattic/jetpack-connection';

const onDisconnectedCallback = useCallback( () => alert( 'Successfully Disconnected' ) );

<ConnectionStatusCard
	apiRoot={ APIRoot }
	apiNonce={ APINonce }
	isRegistered={ true }
	isUserConnected={ true }
	redirectUri="tools.php?page=wpcom-connection-manager"
/>
```

#### Jetpack product discussion
p1HpG7-cl4-p2

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
If you're using a JN site, you need to install [this plugin](https://gist.github.com/sergeymitr/bc005e1b893e778c03e4f43debb91c30) to activate the Connection Manager page.

1. Fully connect Jetpack.
2. Go to "Tools -> Connection Manager".
3. Confirm that you see the following info regarding your connection:
<img width="470" alt="Screenshot 2021-07-22 at 5 34 10 PM" src="https://user-images.githubusercontent.com/1758399/126662455-a8b542e4-6b4f-46e8-afeb-16c92970d49d.png">
4. Click the link to Disconnect
5. Confirm the ConnectScreen component will appear when you successfully disconnect.
6. Connect only at a site level (aka do not connect your WordPress.com account - This can be achieved by navigating back after you are redirected to WordPress.com)
7. Make sure you see the following info regarding your connection:
<img width="516" alt="Screenshot 2021-07-22 at 5 34 53 PM" src="https://user-images.githubusercontent.com/1758399/126662811-97eee89d-8550-4afc-ae3d-12d41cffdd8a.png">
8. Click the "Connect your WordPress.com account" button
9. Verify that you are redirected to WordPress.com to connect your account and then successfully back to the Connection Manager
10. At this stage you should be seeing the same info with Step 3

Important note: When testing with a non-logged in WordPress.com user and choose to not connect your account the flow gets very messy and is probably broken. This is a known issue (cc @leogermani to double check) out of this PR's scope.
